providers/openlineage/src/airflow/providers/openlineage/plugins/macros.py

```
@@ -58,15 +59,27 @@ def lineage_run_id(task_instance: TaskInstance):
         For more information take a look at the guide:
         :ref:`howto/macros:openlineage`
     """
-    if hasattr(task_instance, "logical_date"):
-        logical_date = task_instance.logical_date
+    if AIRFLOW_V_3_0_PLUS:
+        context = task_instance.get_template_context()
+        if hasattr(task_instance, "dag_run"):
+            dag_run = task_instance.dag_run
+        elif hasattr(context, "dag_run"):
+            dag_run = context["dag_run"]
+        else:
+            return ""
```

Should we silently return an empty string here? What is the scenario here where there is no dag_run? Can this happen?

---

>Should we silently return an empty string here? What is the scenario here where there is no dag_run? Can this happen?

AFAIK it can happen in tests only.

---

Is there anyway we can provide this information in the test cases (either dag_run attr or context)? Not sure if it's a good idea to add a logic in the code, that if executed, breaks the result of this macro, purely for the sake of making tests pass.

---

Removed the case.